sphinx_doc_src/fish_sphinxext.py

```diff
@@ -0,0 +1,82 @@
+import os.path
+import re
+
+from docutils import nodes
+from docutils.parsers import rst
+from docutils.statemachine import ViewList
+
+STRING_DOC = os.path.join(os.path.dirname(__file__), "cmds/string.rst")
```

Comment:
I think the naming overdoes it a bit with `string` everywhere. It's entirely plausible we'll split more docs in future.